### PR TITLE
[1.18] make lack of hugetlb support non-fatal for cgroupv1

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
lack of hugetlb support in the kernel should not be a fatal error in cgroupv1.

this PR patches the error in 1.18. https://github.com/cri-o/cri-o/pull/3565 will fix it in master

#### Which issue(s) this PR fixes:

fixes https://github.com/cri-o/cri-o/issues/3717
#### Does this PR introduce a user-facing change?

```release-note
Fix a bug where CRI-O could not start a container if CONFIG_CGROUP_HUGETLB was not set in the kernel
```

### CODE REVIEW

File: server/container_create_linux.go [hunk 1]
```diff
@@ -58,17 +57,15 @@ var (
 
 // cgroupv1HasHugetlb returns whether the hugetlb controller is present on
 // cgroup v1.
-func cgroupv1HasHugetlb() (bool, error) {
+func cgroupv1HasHugetlb() bool {
 	_cgroupv1HasHugetlbOnce.Do(func() {
 		if _, err := ioutil.ReadDir("/sys/fs/cgroup/hugetlb"); err != nil {
-			_cgroupv1HasHugetlbErr = errors.Wrap(err, "readdir /sys/fs/cgroup/hugetlb")
 			_cgroupv1HasHugetlb = false
```
Comment (TomSweeneyRedHat) [inline server/container_create_linux.go:63]: IDK, but might it be useful to have a debug log print out here if we set this to false?

**Comment by TomSweeneyRedHat** [comment]
One questions, otherwise LGTM with happy tests.

File: server/container_create_linux.go (hunk 1 above)
Comment (haircommander) [reply server/container_create_linux.go:63]: good idea, added

Comment (haircommander) [reply server/container_create_linux.go:63]: ach, nevermind, the function call flow doesn't make this simple or clear. I think we're okay with out.

**Comment by haircommander** [comment]
PTAL @saschagrunert @mrunalp @umohnani8

**Pull request merged**